Code/GraphMol/AddHs.cpp

```diff
@@ -855,6 +909,38 @@ bool shouldRemoveH(const RWMol &mol, const Atom *atom,
   return removeIt;
 }
 
+// Do not remove H atoms that are part of SGroups that only contain H atoms.
```

Comment:
If this is something we don't expect to see, is it worth guarding against?

Reply:
I think it is. I'd rather come back to this and remove those in the future, when/if we know it's safe removing them, than make someone lose information just because we haven't seen any of these.